lib/proxy.js

```diff
@@ -55,8 +55,15 @@ class Gateway {
         }
       })
       .catch(errors.RequestError, reason => {
-        logger.Error('Request to the server failed.', { exit: false });
-        logger.Error(`${reason.cause}`, { exit: false });
+        switch (reason.cause.code) {
+          case "ENOTFOUND":
+            ServerError.addressNotFound(reason);
+            break;
+
+          default:
```

Comment:
can we also include useful error message for handling in the scenario that our stack is currently down / in the maintenance - `ECONNREFUSED` ? We could say something like "We are temporarily unavailable, please re-try in few seconds" or something equivalent